### PR TITLE
Add command to clean pictures path when changing instance URL

### LINKED ISSUE #2852: If you change the URL of your wallabag instance, all links to local images are broken
### Issue details

When the option `Download images locally` is on 1 (in `Internal Settings` → `Misc`) and if you change `URL of *your* wallabag instance` , all links to local images are not changed to the new setting, so they are all broken.

### Environment

* wallabag version (or git revision) that exhibits the issue: 2.2.1
* How did you install wallabag? Via `git clone` or by downloading the package? make update
* Last wallabag version that did not exhibit the issue (if applicable):
* php version:
* OS: raspbian
* type of hosting (shared or dedicated):
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): sqlite

### Steps to reproduce/test case

0. Lets say your URL is `http://wallabag.name.tld`
1. In `internal settings` → `misc` → `URL of *your* wallabag instance` 
1bis. Put: `http://toto.name.tld`
2. Add an article
2bis. Notice you have a 404 error with images because links begin with `http://toto.name.tld`
3. Back to `internal settings` → `misc` → `URL of *your* wallabag instance` 
3bis. Put: `http://wallabag.name.tld`
4. Back to the previous article: your images are still 404
4bis: Note links for images are still with `http://toto.name.tld` instead of `http://wallabag.name.tld`

### PR DESCRIPTION
**Pull request by nicosomb**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    |no
| Deprecations? |no
| Tests pass?   | yes/no
| Documentation |no
| Translation   |no
| CHANGELOG.md  |no
| License       | MIT

Should fix #2852. 

Tests missing.

### CODE REVIEW

File: package-lock.json
Comment (j0k3r) [inline package-lock.json:1]: ???

Comment (nicosomb) [reply package-lock.json:1]: 🫣

File: src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $oldUrl = $input->getArgument('old-url');
+
+        $entries = $this->entryRepository->findAll();
```
Comment (yguedidi) [inline src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php:45]: I think loading all entries is not very scalable, especially when there are lot of entries 😅 
maybe use `toIterable()`? https://www.doctrine-project.org/projects/doctrine-orm/en/2.15/reference/batch-processing.html

Comment (nicosomb) [reply src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php]: @yguedidi I made a change about the findAll(). Not sure if it's ok ...

Comment (yguedidi) [reply src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php]: To me it's still hydrating all entries in memory..
I think the exact fix is similar to https://www.doctrine-project.org/projects/doctrine-orm/en/2.15/reference/batch-processing.html#iterating-results
- create a select all QUery
- then use toIterate() in the loop

image

Comment (nicosomb) [reply src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php]: @yguedidi is that better?

File: src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        $io->text('Retrieve existing entries');
+        $i = 1;
+        foreach ($query->toIterable() as $entry) {
+            $content = str_replace($oldUrl, $this->wallabagUrl, $entry->getContent());
+            $entry->setContent($content);
+
+            $previewPicture = str_replace($oldUrl, $this->wallabagUrl, $entry->getPreviewPicture());
+            $entry->setPreviewPicture($previewPicture);
+
+            $this->entityManager->persist($entry);
```
Comment (yguedidi) [inline src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php:55]: not sure the persist is needed, as the entry comes from Doctrine so it's already known by it, but to be checked thanks to the test

**Review by yguedidi — APPROVED** [review]
one remark to check, approve anyway on my side

**Review by yguedidi — APPROVED** [review]

**Review by j0k3r — COMMENTED** [review]
Looks great, you can squash :+1:

**Comment by nicosomb** [comment]
@j0k3r done. But, as a new feature, please don't merge, it's for `2.7.0`.

**Comment by j0k3r** [comment]
I know I know :)

**Review by j0k3r — APPROVED** [review]

**Pull request merged**